DataSource/Cloudwatch: Add Auth Option and Cache Config Per DataSource (#23439)
grafana/grafana: docs/sources/features/datasources/cloudwatch.md

| _Default_                  | Default data source means that it will be pre-selected for new panels.                                  |
| _Default Region_           | Used in query editor to set region (can be changed on per query basis)                                  |
| _Custom Metrics namespace_ | Specify the CloudWatch namespace of Custom metrics                                                      |

> | _Authentication Provider_  | Specify which method Grafana will use to find the AWS credentials.                                      |

Reviewer: ```suggestion
| _Authentication Provider_  | Specify which method Grafana uses to find the AWS credentials.                                      |
```

### AWS credentials

> There are four different authentication providers available. The `AWS SDK

Reviewer: These line breaks are painful. Please fix them so this is easier to read.

if you're [running on an EC2 instance](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html),
[in an ECS task](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-iam-roles.html)
or for a [Service Account in a Kubernetes cluster](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html).
The `Assume Role` authentication provider works similarly, but it will instead use the
`AWS SDK Default` authentication provider to assume the specified role, which it will
then use to query the CloudWatch API.

The `Credentials file` authentication provider corresponds directly to the
[SharedCredentialsProvider](https://docs.aws.amazon.com/sdk-for-go/api/aws/credentials/#SharedCredentialsProvider)
provider in the go sdk. In short, it will read the aws shared credentials file
and find the given profile. While `AWS SDK Default` will also find the shared
credentials file (on account of the SharedCredentialsProvider being part of the
default chain) this option allows you to specify which profile to use without
messing with environment variables. This option will not do any implicit

> fallbacks to other providers. If the SharedCredentialsProvider fails

Reviewer: ```suggestion
fallbacks to other providers. If the SharedCredentialsProvider fails, then
```

authentication will fail.

The `Access & secret key` corresponds to the
[StaticProvider](https://docs.aws.amazon.com/sdk-for-go/api/aws/credentials/#StaticProvider)
and will use the given access key id and the secret key to authenticate. If

> this provider fails for whatever reason the authentication will fail. No

Reviewer: ```suggestion
this provider fails for whatever reason, then the authentication will fail. No
```
